services/wallet/walletconnect/database.go

```diff
@@ -39,65 +67,118 @@ func ChangePairingState(db *sql.DB, topic Topic, active bool) error {
 	}
 
 	return nil
 }
 
-func GetPairingByTopic(db *sql.DB, topic Topic) (*Pairing, error) {
-	querySQL := `SELECT topic, expiry_timestamp, active, app_name, url, description, icon, verified_is_scam, verified_origin, verified_verify_url, verified_validation FROM wallet_connect_pairings WHERE topic = ?`
+func GetSessionByTopic(db *sql.DB, topic Topic) (*DbSession, error) {
+	querySQL := `
+	SELECT *
+	FROM
+		wallet_connect_sessions
+	WHERE
+		topic = ?`
 
 	row := db.QueryRow(querySQL, topic)
 
-	var pairing Pairing
-	err := row.Scan(&pairing.Topic, &pairing.Expiry, &pairing.Active, &pairing.AppName, &pairing.URL, &pairing.Description, &pairing.Icon, &pairing.Verified.IsScam, &pairing.Verified.Origin, &pairing.Verified.VerifyURL, &pairing.Verified.Validation)
+	var session DbSession
+	err := row.Scan(&session.Topic,
+		&session.PairingTopic,
+		&session.Expiry,
+		&session.Active,
+		&session.DappName,
+		&session.DappURL,
+		&session.DappDescription,
+		&session.DappIcon,
+		&session.DappVerifyURL,
+		&session.DappPublicKey)
 	if err != nil {
 		return nil, err
 	}
 
-	return &pairing, nil
+	return &session, nil
 }
 
-// GetActivePairings returns all active pairings (active and not expired) that have an expiry timestamp newer or equal to the given timestamp.
-func GetActivePairings(db *sql.DB, expiryNotOlderThanTimestamp int64) ([]Pairing, error) {
-	querySQL := `SELECT topic, expiry_timestamp, active, app_name, url, description, icon, verified_is_scam, verified_origin, verified_verify_url, verified_validation FROM wallet_connect_pairings WHERE active != 0 AND expiry_timestamp >= ? ORDER BY expiry_timestamp DESC`
+func GetSessionsByPairingTopic(db *sql.DB, pairingTopic Topic) ([]DbSession, error) {
+	querySQL := `
+	SELECT *
+	FROM
+		wallet_connect_sessions
+	WHERE
+		pairing_topic = ?`
 
-	rows, err := db.Query(querySQL, expiryNotOlderThanTimestamp)
+	rows, err := db.Query(querySQL, pairingTopic)
 	if err != nil {
 		return nil, err
 	}
 	defer rows.Close()
 
-	pairings := make([]Pairing, 0, 2)
+	sessions := make([]DbSession, 0, 2)
 	for rows.Next() {
-		var pairing Pairing
-		err := rows.Scan(&pairing.Topic, &pairing.Expiry, &pairing.Active, &pairing.AppName, &pairing.URL, &pairing.Description, &pairing.Icon, &pairing.Verified.IsScam, &pairing.Verified.Origin, &pairing.Verified.VerifyURL, &pairing.Verified.Validation)
+		var session DbSession
+		err := rows.Scan(&session.Topic,
+			&session.PairingTopic,
+			&session.Expiry,
+			&session.Active,
+			&session.DappName,
+			&session.DappURL,
+			&session.DappDescription,
+			&session.DappIcon,
+			&session.DappVerifyURL,
+			&session.DappPublicKey)
 		if err != nil {
 			return nil, err
 		}
 		if err := rows.Err(); err != nil {
 			return nil, err
 		}
-		pairings = append(pairings, pairing)
-	}
-	if err := rows.Err(); err != nil {
-		return nil, err
+		sessions = append(sessions, session)
 	}
 
-	return pairings, nil
+	return sessions, rows.Err()
```

Comment:
If I understand correctly you're saying that this is better:
```
if err := rows.Err(); err != nil {
    return nil, err
}
return sessions, nil
```
than this:
`return sessions, rows.Err()`

But isn't that the choice depends on your preference only cause I see those two the same in the end.

Reply:
> If I understand correctly you're saying that this is better:
I separated the two things and it got confusing I see. The only important thing was to move the check for `rows.Err()` outside the loop.  Two steps approach is more idiomatic so that you don't provide a partial unusable result when an error is returned. However, in practice the one liner should work very well given that the first check after the call should be for the error.